tests fail on Strawberry perl 5.30.0.1

Just tried installing with Strawberry perl 5.30.0.1 PDL version and got this failure.  

It looks very similar to #1 (readline on open file handle), but with a few differences in what is reported.  


```
cpanm -v Test2::Plugin::FauxHomeDir
cpanm (App::cpanminus) 1.7044 on perl 5.030000 built for MSWin32-x64-multi-thread
Work directory is /.cpanm/work/1558657450.7308
You have make C:\berrybrew\5.30.0_64_PDL\c\bin\gmake.exe
You have LWP 6.39
Falling back to Archive::Tar 2.32
Searching Test2::Plugin::FauxHomeDir () on cpanmetadb ...
--> Working on Test2::Plugin::FauxHomeDir
Fetching http://www.cpan.org/authors/id/P/PL/PLICEASE/Test2-Plugin-FauxHomeDir-0.04.tar.gz ... OK
Unpacking Test2-Plugin-FauxHomeDir-0.04.tar.gz
Entering Test2-Plugin-FauxHomeDir-0.04
Checking configure dependencies from META.json
Checking if you have ExtUtils::MakeMaker 6.58 ... Yes (7.36)
Running Makefile.PL
Configuring Test2-Plugin-FauxHomeDir-0.04 ... Checking if your kit is complete...
Looks good
Generating a gmake-style Makefile
Writing Makefile for Test2::Plugin::FauxHomeDir
Writing MYMETA.yml and MYMETA.json
OK
Checking dependencies from MYMETA.json ...
Checking if you have Test2::V0 0.000060 ... Yes (0.000122)
Checking if you have ExtUtils::MakeMaker 0 ... Yes (7.36)
Checking if you have Test2::API 1.302015 ... Yes (1.302164)
Building and testing Test2-Plugin-FauxHomeDir-0.04 ... cp lib/Test2/Plugin/FauxHomeDir.pm blib\lib/Test2/Plugin/FauxHomeDir.pm
"C:\berrybrew\5.30.0_64_PDL\perl\bin\perl.exe" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib\lib', 'blib\arch')" t/*.t
t/00_diag.t ........................................... 1/? #
#
#
# HARNESS_ACTIVE=1
# HARNESS_VERSION=3.42
# PERL5LIB=C:\.cpanm\work\1558657450.7308\Test2-Plugin-FauxHomeDir-0.04\blib\lib;C:\.cpanm\work\1558657450.7308\Test2-Plugin-FauxHomeDir-0.04\blib\arch
# PERL_MM_USE_DEFAULT=1
# PERL_USE_UNSAFE_INC=1
#
#
#
# PERL5LIB path
# C:\.cpanm\work\1558657450.7308\Test2-Plugin-FauxHomeDir-0.04\blib\lib
# C:\.cpanm\work\1558657450.7308\Test2-Plugin-FauxHomeDir-0.04\blib\arch
#
#
#
# perl                5.030000
# ExtUtils::MakeMaker 7.36
# Test2::API          1.302164
# Test2::V0           0.000122
#
#
#
#           Before:
#         HOME = undef
#    HOMEDRIVE = C:
#     HOMEPATH = \Users\shawn
#         USER = undef
#     USERNAME = shawn
#  USERPROFILE = C:\Users\shawn
#
#           After:
#         HOME = undef
#    HOMEDRIVE = C:
#     HOMEPATH = \Users\shawn\AppData\Local\Temp\8o6UPCRMWq\home\shawn
#         USER = undef
#     USERNAME = shawn
#  USERPROFILE = C:\Users\shawn\AppData\Local\Temp\8o6UPCRMWq\home\shawn
#
#
#
t/00_diag.t ........................................... ok
t/01_use.t ............................................ ok
t/test2_plugin_fauxhomedir.t .......................... 1/? readline() on closed filehandle $fh at t/test2_plugin_fauxhomedir.t line 60.
    # Failed test at t/test2_plugin_fauxhomedir.t line 57.
    # Failed test at t/test2_plugin_fauxhomedir.t line 63.
    # +---------+-------+
    # | GOT     | CHECK |
    # +---------+-------+
    # | <UNDEF> | xx\n  |
    # +---------+-------+

# Failed test 'File::HomeDir'
# at t/test2_plugin_fauxhomedir.t line 64.
t/test2_plugin_fauxhomedir.t .......................... Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/6 subtests
t/test2_plugin_fauxhomedir__with_file_homedir_test.t .. ok

Test Summary Report
-------------------
t/test2_plugin_fauxhomedir.t                        (Wstat: 256 Tests: 6 Failed: 1)
  Failed test:  5
  Non-zero exit status: 1
Files=4, Tests=9,  2 wallclock secs ( 0.08 usr +  0.03 sys =  0.11 CPU)
Result: FAIL
Failed 1/4 test programs. 1/9 subtests failed.
gmake: *** [Makefile:856: test_dynamic] Error 255
FAIL
! Installing Test2::Plugin::FauxHomeDir failed. See \.cpanm\work\1558657450.7308\build.log for details. Retry with --force to force install it.
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,24 +1,35 @@
-language: perl
-sudo: false
+language: minimal
+dist: xenial
+services:
+  - docker
+
+before_install:
+  - curl https://raw.githubusercontent.com/plicease/cip/master/bin/travis-bootstrap | bash
+  - cip before-install
 
 install:
-  - perlbrew list
-  - cpanm -n Dist::Zilla
-  - dzil authordeps --missing | cpanm -n
-  - dzil listdeps   --missing | cpanm -n
-
-perl:
-  - "5.14"
-  - "5.16"
-  - "5.18"
-  - "5.20"
-  - "5.22"
-  - "5.24"
-  - "5.26"
+  - cip diag
+  - cip install
 
 script:
-  - dzil test -v
+  - cip script
 
-env:
-  global:
-    - PERL_USE_UNSAFE_INC=0
+jobs:
+  include:
+    - env: CIP_TAG=5.30
+    - env: CIP_TAG=5.29
+    - env: CIP_TAG=5.28
+    - env: CIP_TAG=5.26
+    - env: CIP_TAG=5.24
+    - env: CIP_TAG=5.22
+    - env: CIP_TAG=5.20
+    - env: CIP_TAG=5.18
+    - env: CIP_TAG=5.16
+    - env: CIP_TAG=5.14
+    - env: CIP_TAG=5.12
+    - env: CIP_TAG=5.10
+    - env: CIP_TAG=5.8
+
+cache:
+  directories:
+    - "$HOME/.cip"

--- a/Changes
+++ b/Changes
@@ -1,7 +1,9 @@
 Revision history for {{$dist->name}}
 
 {{$NEXT}}
-  - Patch the patched File::HomeDir that is patched by Portable Strawberry Perl on windows.
+  - Patch the patched File::HomeDir that is patched by Portable Strawberry Perl on windows
+    (gh#2, gh#4)
+  - Better diagnostics when things fail (gh#3 shawnlaffan++)
 
 0.04      2018-04-18 12:46:06 -0400
   - Add real_home_dir method

--- a/Changes
+++ b/Changes
@@ -1,6 +1,7 @@
 Revision history for {{$dist->name}}
 
 {{$NEXT}}
+  - Patch the patched File::HomeDir that is patched by Portable Strawberry Perl on windows.
 
 0.04      2018-04-18 12:46:06 -0400
   - Add real_home_dir method

--- a/README.md
+++ b/README.md
@@ -81,7 +81,11 @@ test is passing) will be raised at the end of the test if you attempt this.
 
 # AUTHOR
 
-Graham Ollis <plicease@cpan.org>
+Author: Graham Ollis <plicease@cpan.org>
+
+Contributors:
+
+Shawn Laffan (SLAFFAN)
 
 # COPYRIGHT AND LICENSE
 

--- a/author.yml
+++ b/author.yml
@@ -7,6 +7,8 @@ pod_spelling_system:
   # intentionally
   stopwords:
     - msys
+    - SLAFFAN
+    - Laffan
 
 pod_coverage:
   skip: 0

--- a/dist.ini
+++ b/dist.ini
@@ -32,6 +32,8 @@ remove = if
 remove = File::Spec
 remove = File::Temp
 remove = File::Path
+remove = Portable
+remove = File::HomeDir
 
 ;[Prereqs]
 ;Foo::Bar = 0

--- a/dist.ini
+++ b/dist.ini
@@ -35,9 +35,9 @@ remove = File::Path
 remove = Portable
 remove = File::HomeDir
 
-;[Prereqs]
-;Foo::Bar = 0
-
 [Author::Plicease::Upload]
 cpan = 1
 
+[Author::Plicease::Thanks]
+current = Graham Ollis <plicease@cpan.org>
+contributor = Shawn Laffan (SLAFFAN)

--- a/lib/Test2/Plugin/FauxHomeDir.pm
+++ b/lib/Test2/Plugin/FauxHomeDir.pm
@@ -104,6 +104,7 @@ hiding bugs.
 my $real;
 my $faux;
 my $user;
+my @mocks;
 
 sub real_home_dir
 {
@@ -112,6 +113,8 @@ sub real_home_dir
 
 sub import
 {
+  my @notes;
+
   unless(defined $faux)
   {
     if($^O eq 'MSWin32')
@@ -146,6 +149,22 @@ sub import
     {
       $ENV{USERPROFILE} = $faux;
       ($ENV{HOMEDRIVE}, $ENV{HOMEPATH}) = File::Spec->splitpath($faux,1);
+      if(eval { require Portable })
+      {
+        push @notes, "Portable strawberry detected";
+        if(eval { require File::HomeDir })
+        {
+          push @notes, "Patching File::HomeDir";
+          require File::HomeDir;
+          require Test2::Mock;
+          push @mocks, Test2::Mock->new(
+            class => 'File::HomeDir',
+            override => [
+              my_home => sub { $faux },
+            ],
+          );
+        }
+      }
     }
     elsif($^O eq 'cygwin')
     {
@@ -157,6 +176,9 @@ sub import
     {
       $ENV{HOME} = $faux;
     }
+
+    push @notes, "Test2::Plugin::FauxHomeDir using faux home dir $faux";
+    push @notes, "Test2::Plugin::FauxHomeDir real home dir is    $real";
     
     test2_add_callback_post_load(sub {
       test2_stack()->top;
@@ -165,7 +187,7 @@ sub import
         Test2::Event::Note->new(
           message => $_
         ),
-      ) for "Test2::Plugin::FauxHomeDir using faux home dir $faux", "Test2::Plugin::FauxHomeDir real home dir is    $real";
+      ) for @notes;
     });
 
     test2_add_callback_exit(sub {

--- a/lib/Test2/Plugin/FauxHomeDir.pm
+++ b/lib/Test2/Plugin/FauxHomeDir.pm
@@ -154,8 +154,11 @@ sub import
         push @notes, "Portable strawberry detected";
         if(eval { require File::HomeDir })
         {
+          # annoyingly, Strawberry Portable Perl patches
+          # File::HomeDir, but not things like File::Glob
+          # so since there isn't a good interface to override
+          # this behavior, we need to patch the patch :(
           push @notes, "Patching File::HomeDir";
-          require File::HomeDir;
           require Test2::Mock;
           push @mocks, Test2::Mock->new(
             class => 'File::HomeDir',

--- a/maint/cip-before-install
+++ b/maint/cip-before-install
@@ -1,0 +1,3 @@
+#!/bin/bash
+
+cip exec cpanm -n File::HomeDir Path::Tiny

--- a/t/00_diag.t
+++ b/t/00_diag.t
@@ -12,6 +12,7 @@ my $post_diag;
 $modules{$_} = $_ for qw(
   ExtUtils::MakeMaker
   Test2::API
+  Test2::Mock
   Test2::V0
 );
 

